File: test/ping.js
#!/usr/bin/env node --no-warnings

//
// ping with node
//

'use strict';
const postData = async () => {
  try {
    const data = {
      name: 'John Doe',
      email: 'johndoe@example.com'
    };

    const response = await fetch('http://piedmont:9890');

    const result = await response.json();

    console.log(result);
  } catch (error) {
    console.error('Error:', error);
  }
};

postData();

File: test/ping.py
#!/usr/bin/env python3
# dpw@plaza.localdomain
# 2023-06-14 17:15:17

# best to use node to test

# import argparse
import http.client

host = "piedmont:9890"

def ping():
    conn = http.client.HTTPConnection(host)
    conn.request("GET", "/")

    response = conn.getresponse()
    print(response.status, response.reason)

    if response.reason == "OK":
        body = response.read()
        print(body)

    conn.close()

def main():
    ping()

if __name__ == '__main__':
    main()
